bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/connector/EpsonProjectorSerialConnector.java

```diff
@@ -0,0 +1,197 @@
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void disconnect() throws EpsonProjectorException {
+        if (out != null) {
+            logger.debug("Close serial out stream");
+            IOUtils.closeQuietly(out);
+            out = null;
+        }
+        if (in != null) {
+            logger.debug("Close serial in stream");
+            IOUtils.closeQuietly(in);
+            in = null;
+        }
+        if (serialPort != null) {
+            logger.debug("Close serial port");
+            serialPort.close();
+            serialPort.removeEventListener();
+            serialPort = null;
+        }
+
+        logger.debug("Closed");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String sendMessage(String data, int timeout) throws EpsonProjectorException {
+        if (in == null || out == null) {
+            connect();
+        }
+
+        try {
+            // flush input stream
+            if (in.markSupported()) {
+                in.reset();
+            } else {
+
+                while (in.available() > 0) {
+
+                    int availableBytes = in.available();
+
+                    if (availableBytes > 0) {
+
+                        byte[] tmpData = new byte[availableBytes];
+                        in.read(tmpData, 0, availableBytes);
+                    }
+                }
+            }
+
+            return sendMmsg(data, timeout);
+
+        } catch (IOException e) {
+
+            logger.debug("IO error occurred...reconnect and resend ones");
+            disconnect();
+            connect();
+
+            try {
+                return sendMmsg(data, timeout);
+            } catch (IOException e1) {
+                throw new EpsonProjectorException(e);
+            }
+
+        } catch (Exception e) {
+            throw new EpsonProjectorException(e);
+        }
+    }
+
+    @Override
+    public void serialEvent(SerialPortEvent arg0) {
+        try {
+            logger.trace("RXTX library CPU load workaround, sleep forever");
+            Thread.sleep(Long.MAX_VALUE);
+        } catch (InterruptedException e) {
+        }
+    }
+
+    private String sendMmsg(String data, int timeout) throws IOException, EpsonProjectorException {
+        out.write(data.getBytes());
+        out.write("\r\n".getBytes());
+        out.flush();
+
+        String resp = "";
+
+        long startTime = System.currentTimeMillis();
+        long elapsedTime = 0;
+
+        while (elapsedTime < timeout) {
+            int availableBytes = in.available();
+            if (availableBytes > 0) {
+                byte[] tmpData = new byte[availableBytes];
+                int readBytes = in.read(tmpData, 0, availableBytes);
+                resp = resp.concat(new String(tmpData, 0, readBytes));
+
+                if (resp.contains(":")) {
+                    return resp;
+                }
+            } else {
+                try {
+                    Thread.sleep(100);
```

Comment:
Can we do without this sleep or use a scheduler?

Reply:
This would probably need to be extensively tested since it might be there for a reason, I'm a little bit wary of touching this working code.